architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md

```diff
@@ -0,0 +1,230 @@
+## Authoring a User-Defined-Type schema
+
+* **Author**: Reshma Abdul Rahim (@reshrahim)
+
+## Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+User-Defined Types enable Platform engineers to define and deploy their organization's custom services in Radius. More details around the feature, user scenarios and end-end-experience can be found [here](/architecture/2024-06-resource-extensibility-feature-spec.md). This feature specification document details on the requirements and the user experience for authoring and updating the schema for the User-Defined-Type.
+
+### Top level goals
+<!-- At the most basic level, what are we trying to accomplish? -->
+
+- Enable a simple and intuitive experience for platform engineers to define the schema for their User-Defined-Type with minimal lines of code
+- Ensure that there is low concept count for the user to understand and get started with defining the schema for their User-Defined-Type
+- Ensure that users have clear visibility into the validation process of the schema definition to guarantee its correctness.
+
+### Non-goals (out of scope)
+<!-- What are we explicitly not trying to accomplish? -->
+These goals are out of scope for the first iteration but might be considered in future iterations based on user feedback.
+
+- Providing a full-fledged TypeSpec tooling experience for authoring the schema definition with autocompletion, error validation at development time.
+- Advanced scenarios defining the capabilities of the resource type in the schema definition. This includes defining capabilities like connections to other resource, output resources or infrastructure resources produced by the resource type etc.
+- Defining child resources or nested resources in the schema definition
+- Extensibility on CRUDL operations for the resource type
+- Providing solutions to handle breaking changes on API versions of the resource type (?)
```

Comment:
Can we highlight which of these are out of scope forever vs out of scope for the first cut? I think this could send the message that some of these ideas aren't important, and they obviously are.

Reply:
I have separated extensibility on CRUDL operations as a goal out of scope forever.